front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -0,0 +1,196 @@
+  if (isRowVia) {
+    return (
+      <button type="button" onClick={removeVia}>
+        ❌
+      </button>
+    );
+  }
+  return <div />;
+};
+
+const TimesStops = ({ pathProperties, pathSteps = [], setPathProperties }: TimesStopsProps) => {
+  const { t } = useTranslation('timesStops');
+  const dispatch = useAppDispatch();
+  const { upsertViaFromSuggestedOP, updatePathSteps } = useOsrdConfActions();
+
+  const [timesStopsSteps, setTimesStopsSteps] = useState<PathWaypointColumn[]>([]);
+
+  useEffect(() => {
+    const suggestedOPs = formatSuggestOPsToRowOPs(pathProperties.allVias, t);
+    setTimesStopsSteps(suggestedOPs);
+  }, [pathSteps]);
+
+  const columns: Column<PathWaypointColumn>[] = useMemo(
+    () => [
+      {
+        ...keyColumn<PathWaypointColumn, 'name'>('name', createTextColumn()),
+        title: t('name'),
+        disabled: true,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'ch'>('ch', createTextColumn()),
+        title: 'Ch',
+        disabled: true,
+        grow: 0.1,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'arrival'>('arrival', timeColumn),
+        title: t('arrivalTime'),
+
+        // We should not be edit the arrival time of the origin
+        cellClassName: ({ rowIndex }) => (rowIndex === 0 ? 'dsg-hidden-cell' : ''),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'stopFor'>(
+          'stopFor',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+          })
+        ),
+        title: `${t('stopTime')}`,
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'onStopSignal'>(
+          'onStopSignal',
+          checkboxColumn as Partial<Column<boolean | undefined>>
+        ),
+        title: t('receptionOnClosedSignal'),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'theoreticalMargin'>(
+          'theoreticalMargin',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+            placeholder: t('theoreticalMarginPlaceholder'),
+            formatBlurredInput: (value) => {
+              if (!value || value === 'none') return '';
+              if (!marginRegExValidation.test(value)) {
+                return `${value}${t('theoreticalMarginPlaceholder')}`;
+              }
+              return value;
+            },
+          })
+        ),
+        cellClassName: ({ rowData }) => (rowData.isMarginValid ? '' : 'invalidCell'),
+        title: t('theoreticalMargin'),
+        disabled: ({ rowIndex }) => rowIndex === findLastIndex(pathProperties.allVias),
+      },
+    ],
+    [t, pathProperties, timesStopsSteps]
+  );
+
+  const removeVia = (
+    rowData: PathWaypointColumn,
+    setRowData: (rowData: PathWaypointColumn) => void
+  ) => {
+    const updatedPathSteps = [...pathSteps];
+    const index = compact(updatedPathSteps).findIndex((step) => {
```

Comment:
compact creates a new array, so you don't need to duplicate the pathSteps
```suggestion
    const index = compact(pathSteps).findIndex((step) => {
```

Reply:
Done, thank you for this explanation.